guacamole/src/main/java/org/apache/guacamole/rest/directory/DirectoryObjectTranslator.java

```diff
@@ -145,8 +145,8 @@ public Map<String, String> filterAttributes(Collection<Form> schema,
                 String attributeValue = attributes.get(attributeName);
 
                 // Include attribute value within filtered map only if
-                // (1) defined and (2) present within provided map
-                if (attributeValue != null)
+                // present within provided map
+                if (attributes.containsKey(attributeName))
```

Comment:
Suggest `attributeValue != null || attributes.containsKey(attributeName)` to avoid the double lookup except where necessary.